nil-pointer in getTestSigChan

After upgrading to the latest digest I am seeing: 
```bash
panic: interface conversion: interface {} is nil, not <-chan os.Signal

goroutine 108 [running]:
github.com/oklog/run.getTestSigChan(...)
	/go/pkg/mod/github.com/oklog/run@v1.1.1-0.20231129054641-daf88ede3394/actors.go:51
github.com/oklog/run.SignalHandler.func1()
	/go/pkg/mod/github.com/oklog/run@v1.1.1-0.20231129054641-daf88ede3394/actors.go:31 +0x355
github.com/oklog/run.(*Group).Run.func1({0xc0006a1710?, 0xc000690a50?})
	/go/pkg/mod/github.com/oklog/run@v1.1.1-0.20231129054641-daf88ede3394/group.go:38 +0x29
created by github.com/oklog/run.(*Group).Run in goroutine 1
	/go/pkg/mod/github.com/oklog/run@v1.1.1-0.20231129054641-daf88ede3394/group.go:37 +0x67 
```

`getTestSigChan` is only used within tests, but causes an nil-pointer if used outside.

I was able to reproduce the nil-pointer: https://go.dev/play/p/JFXak3UIrB6
From what I can tell there needs to be a nil check on the value retrieved from the context before the type assertion.

--- a/actors.go
+++ b/actors.go
@@ -48,7 +48,8 @@ func SignalHandler(ctx context.Context, signals ...os.Signal) (execute func() er
 type testSigChanKey struct{}
 
 func getTestSigChan(ctx context.Context) <-chan os.Signal {
-	return ctx.Value(testSigChanKey{}).(<-chan os.Signal) // can be nil
+	c, _ := ctx.Value(testSigChanKey{}).(<-chan os.Signal) // can be nil
+	return c
 }
 
 func putTestSigChan(ctx context.Context, c <-chan os.Signal) context.Context {

--- a/actors_test.go
+++ b/actors_test.go
@@ -57,3 +57,11 @@ func TestSignalError(t *testing.T) {
 		t.Errorf("errors.Is(err, ErrSignal): want %v, have %v", want, have)
 	}
 }
+
+func TestSignalHandlerNil(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	var rg Group
+	rg.Add(SignalHandler(ctx, os.Interrupt))
+	cancel()
+	t.Logf("%v", rg.Run())
+}

